larray/core/session.py

```diff
@@ -1602,6 +1617,17 @@ def _check_key_value(self, key, value):
                     msg = str(error).replace("incompatible axes:", "incompatible axes for array '{}':".format(key))\
                         .replace("vs", "was declared as")
                     raise ValueError(msg)
+        else:
+            def equal(v1, v2):
+                if isinstance(v1, (Group, Axis)):
+                    return v1.equals(v2)
+                elif isinstance(v1, Array):
+                    return v1.equals(v2, nans_equal=True)
+                else:
+                    return v1 == v2
+            if not equal(value, attr_def):
+                warnings.warn(f"Cannot modify the value of the variable '{key}' declared as a constant in the "
```

Comment:
You need to **either** change the message (which sounds like the action was blocked) or raise an exception instead of using a warning.

Reply:
I first tried to raise an exception but it breaks the ``__init__`` and the ``load`` methods. 
I think raising an exception is the proper way to do it but I am not sure how to bypass this check when calling the ``__init__`` or ``load`` method?